models/articles-models.js

```diff
@@ -0,0 +1,11 @@
+const db = require('../db/connection');
+
+exports.fetchArticleById = (article_id) => {
+  return db.query('SELECT * FROM articles WHERE article_id = $1;', [article_id]).then((result) => {
+    const [article] = result.rows;
+    if (!article) {
+      throw { status: 404, msg: 'Article not found' };
+    }
```

Comment:
You want this to get thrown into the custom error handling, using Promise.reject, again have a look at the notes: [https://notes.northcoders.com/courses/js-back-end/error-handling](url)